client-js/app/elements/labrad-plot.ts

```diff
@@ -811,3 +789,66 @@ function prettyNumber(value: number, min: number, max: number, res: number): str
   }
 }
 
+
+/**
+ * Get a color for the given value on a scale with the given min and max
+ * limits (zMin <= z <= zMax).
+ */
+function getColor(z: number, zMin: number, zMax: number) {
+  var index = 128;
+  if (zMin !== zMax) {
+    index = Math.floor(255 * (z - zMin) / (zMax - zMin));
```

Comment:
I think a simple alternative to appending the last color a second time to the list is to add an explicit check for `z == zMax`:
```typescript
function getColor(z: number, zMin: number, zMax: number) {
  let index = 128;
  if (zMin !== zMax) {
    if (z == zMax) {
      index = 255;
    } else {
      index = Math.floor(256 * (z - zMin) / (zMax - zMin));
    }
  }
  return COLOR_MAP[index];
}
```